### PR TITLE
Frontend cleanup

### PR DESCRIPTION
**Pull request by nikodem-bilczewski**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,4 @@ dist-ssr
 *.njsproj
 *.sln
 *.sw?
+.env
```
Comment (DanielKlys7) [inline .gitignore]: Not a huge problem but missing empty line, it's an old history decision in C https://gcc.gnu.org/legacy-ml/gcc/2003-11/msg01568.html, good to know tho

File: .prettierrc [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
-	"trailingComma": "all",
-	"jsxSingleQuote": true,
-	"bracketSpacing": true
+  "semi": false,
+  "tabWidth": 2,
+  "printWidth": 80,
+  "importOrder": [
+    "^services(.*)$",
+    "^components/(.*)$",
+    "^assets/(.*)$",
```
Comment (DanielKlys7) [inline .prettierrc]: Not sure bout you, but I'd love to have `services, components, assets` grouped in my imports and not separated which is done in your project by `importOrderSeparation` which is good. Think about creating one regex to match them all.

File: src/components/atoms/Button/Button.types.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { ReactNode } from 'react'
+
+export type ButtonProps = {
+  children: ReactNode
+  cartButton?: boolean
+  primary?: boolean
+  secondary?: boolean
+}
```
Comment (DanielKlys7) [inline src/components/atoms/Button/Button.types.ts]: It isn't an atom actually, regarding your types. It seems like it's molecule at least. Atoms are supposed to be used in larger parts of application and can be fully parametrised. `cartButton` indicates that it only can have cart icon, what this component should take as a props is simply `icon`.

File: src/components/atoms/Button/index.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import classNames from 'classnames'
+
+import { ButtonProps } from 'components/atoms/Button/Button.types'
```
Comment (DanielKlys7) [inline src/components/atoms/Button/index.tsx]: Here you can import relatively regarding the fact that it's really near, actually it's a matter of experience to know when to import absolutely and when to import relatively 🤔

File: src/components/atoms/Button/index.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    },
+    {
+      'w-full bg-lime-700 py-3 text-2xl text-white hover:bg-opacity-80':
+        cartButton,
+    },
+  )
+  return <button className={classes}>{children}</button>
+}
+
+export default Button
```
Comment (DanielKlys7) [inline src/components/atoms/Button/index.tsx]: It looks good, I would only suggest putting the thing that you declare as `classes` directly into components `className`

File: src/components/atoms/IconShoppingCart/index.tsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import ShoppingCartIcon from '@heroicons/react/24/outline/ShoppingCartIcon'
+
+const IconShoppingCart = () => (
+  <div className='rounded-full bg-black bg-opacity-10 p-3 hover:bg-opacity-20'>
+    <ShoppingCartIcon className='h-8 w-8 lg:h-10 lg:w-10' />
+  </div>
+)
+
+export default IconShoppingCart
```
Comment (DanielKlys7) [inline src/components/atoms/IconShoppingCart/index.tsx]: You will end up having lots of icons, these could kinda disturb your ability to locate rest of your atoms on a first sight in directory, what do you think of separating these into `atoms/icons`? 😅

Comment (DanielKlys7) [reply src/components/atoms/IconShoppingCart/index.tsx]: And also is it good idea to create this particullar component over again? What I mean is you could simply do it like this:

```
const IconWrapper = (children) => (
  <div className='rounded-full bg-black bg-opacity-10 p-3 hover:bg-opacity-20'>
    {children}
  </div>
)
```

and then use it:

`<IconWrapper> <ShoppingCartIcon className='h-8 w-8 lg:h-10 lg:w-10' /></IconWrapper>`

this way the only component that we need is in fact the wrapper

File: src/components/molecules/Card/Card.types.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export type CardPropType = {
+  src: string
+  heading: string
+  btnText?: string
+  path?: string
+}
```
Comment (DanielKlys7) [inline src/components/molecules/Card/Card.types.ts]: Uncle Bob recommended avoiding using types in variable names. This includes using "PropType" in the name of a variable that is a prop type, such as "CardPropType."

Using the type in the variable name is redundant and doesn't provide any additional information beyond what is already known about the variable's type. Instead, Uncle Bob suggests using a name that describes the purpose or use of the prop type, such as "CardProps" or "CardConfiguration."

This makes the code easier to read and understand for other developers who may be working on the codebase in the future, as they can quickly grasp the purpose of the prop type based on its name, rather than having to deduce it from the name and type of the variable.

In summary, while using the type in the name of a prop type variable is not necessarily an antipattern, it is generally considered good practice to use descriptive and meaningful variable names that do not include the type of the variable.

I belive that sticking to `CardProps`, or `AntythingProps` in the future would be a good practice for us, what do you think? 🤔

File: src/components/molecules/FooterInfo/useFooterInfo.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { useState } from 'react'
+
+const useFooterInfo = () => {
+  const [footerInfo] = useState<string[]>([
+    'Terms and conditions',
+    'Privacy policy',
+    'Delivery and payment methods',
+    'Loyalty program',
+  ])
+  return footerInfo
```
Comment (DanielKlys7) [inline src/components/molecules/FooterInfo/useFooterInfo.ts]: For a current moment it simply can be `const` :)

File: src/components/molecules/NavigationButtons/index.tsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import NavigationLink from 'components/atoms/NavigationLink'
+import useNavigationButtons from 'components/molecules/NavigationButtons/useNavigationButtons'
```
Comment (DanielKlys7) [inline src/components/molecules/NavigationButtons/index.tsx]: these absolute imports looks way better than relative ones :D

File: src/components/molecules/FooterInfo/useFooterInfo.ts (hunk 8 above)
Comment (DanielKlys7) [reply src/components/molecules/FooterInfo/useFooterInfo.ts]: Also, you don't really have to type variables as simple as this one, typescript is really good in infering those types from the usage :D

File: src/components/molecules/NavigationButtons/useNavigationButtons.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+import { NavButtonType } from 'components/molecules/NavigationButtons/navButton.types'
+
+const useNavigationButtons = (): NavButtonType[] => {
+  const [navButtons] = useState<NavButtonType[]>([
+    { navigateTo: '', btnText: 'Home' },
+    { navigateTo: 'products', btnText: 'Products' },
+    { navigateTo: 'events', btnText: 'Events' },
+    { navigateTo: 'newsletter', btnText: 'Newsletter' },
+  ])
+  return navButtons
```
Comment (DanielKlys7) [inline src/components/molecules/NavigationButtons/useNavigationButtons.ts]: same as in `useFooterInfo`

File: src/components/pages/Products/useProducts.ts
Comment (DanielKlys7) [inline src/components/pages/Products/useProducts.ts:1]: lovely hook, that's the way you should do it

File: src/components/pages/Products/useProducts.ts [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    queryFn: () => getProducts(currentPage),
+    keepPreviousData: true,
+  })
+
+  const products = data?.list
+  const totalPages = data?.meta.pages
+
+  const onPageChange = (pageNumber: number) => {
+    setCurrentPage(pageNumber + 1)
+    window.scrollTo({ top: 0, behavior: 'smooth' })
```
Comment (DanielKlys7) [inline src/components/pages/Products/useProducts.ts:20]: doesn't mean for you to change it, but it may cause some problems https://caniuse.com/?search=scroll-behavior%3A%20smooth

File: src/services/products.ts [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import axios from 'axios'
+
+import { ProductResponse } from 'components/pages/Products/Products.types'
+
+const productsAPI = axios.create({
+  baseURL: import.meta.env.VITE_BASE_URL,
+})
```
Comment (DanielKlys7) [inline src/services/products.ts]: the instance should be contained in your project's config directory as all of our services will use it :D

File: src/services/products.ts [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+
+export const getProducts = async (page: number): Promise<ProductResponse> => {
+  const response = await productsAPI.get(`api/products?perPage=6&page=${page}`)
```
Comment (DanielKlys7) [inline src/services/products.ts]: Also perPage should be parametrized and it's props value should be set as default on what we want. In that case `6`, so

`async (page: number, perPage: number)`

**Review by DanielKlys7 — APPROVED** [review]
